### DISCUSSION TITLE
Question: how is `app` folder structured?

### DISCUSSION THREAD

**Discussion by elie222** [dubinc/dub#237] [General]
Hey,

I noticed the app folder includes dub.sh and [domain].

What stops everything going to the [domain] folder? When is dub.sh used by Next.js?

**Comment by steven-tey** [comment]
Middleware handles all the routing magic 🪄 

Specifically these lines: https://github.com/steven-tey/dub/blob/main/middleware.ts#L50-L57

**Comment by ugniussilevicius** [comment]
Hello, could someone please help me understand how this routing mechanism works? I'm particularly interested in understanding how the directories [domain] and [dub.sh] play a role in the routing process. Currently, when I click on "/pricing", it redirects me back to "/", and the same issue occurs with "/changelog" when running the project locally.

I would highly appreciate it if someone could take the time to provide a detailed explanation of this routing process or how to actually navigate to "/pricing" or "/changelog" in localhost. 

Not only would it help me, but I believe it would also be beneficial for other new developers who might encounter similar challenges.
